### PR TITLE
Use one thread for namespaced runtimes

### PR DESCRIPTION
**Pull request by aviramha**
Not sure if it matters a lot, but if tests pass might worth merging it.

### CODE REVIEW

**Review by eyalb181 — APPROVED** [review]

**Comment by infiniteregrets** [comment]
quick question, what is the performance benefit here? any benchmarks for using the current thread vs multithreaded?

**Comment by aviramha** [comment]
> quick question, what is the performance benefit here? any benchmarks for using the current thread vs multithreaded?

No benchmark, threads have overhead and should be avoided in async if not needed..

File: changelog.d/+internal-single-thread-agent.internal.md
Comment (t4lz) [inline changelog.d/+internal-single-thread-agent.internal.md:1]: This can link to #1287.

Comment (aviramha) [reply changelog.d/+internal-single-thread-agent.internal.md:1]: Thanks!

**Comment by aviramha** [comment]
Closes #1287

**Pull request merged**